lib/praxis/resource_definition.rb

```diff
@@ -11,7 +11,12 @@ module ResourceDefinition
       @version = 'n/a'.freeze
       @actions = Hash.new
       @responses = Hash.new
-      @action_defaults = Trait.new
+
+      # Ensure we inherit any base params defined in the API definition for this version
+      base_attrs = if base_params = ApiDefinition.instance.info.base_params
```

Comment:
If we only use base_attrs here, and get it from the ApiDefinition state, can't we determine it in the 
generate_defaults_block?